### PR TITLE
gcc9 compilation fixes

### LINKED ISSUE #3119: GCC 9: new warnings for -Werror=address-of-packed-member option
Using current GCC trunk, one can see multiple errors:

```
[   79s] /home/abuild/rpmbuild/BUILD/ucx-1.4.0/src/ucs/datastruct/queue.h:269:22: error: taking address of packed member of 'struct <anonymous>' may result in an unaligned pointer value [-Werror=address-of-packed-member]
[   79s]   269 |     ucs_container_of(*iter, typeof(*elem), member)
[   79s] /home/abuild/rpmbuild/BUILD/ucx-1.4.0/src/ucs/sys/compiler_def.h:119:32: note: in definition of macro 'ucs_container_of'
[   79s]   119 |     ( (_type*)( (char*)(void*)(_ptr) - ucs_offsetof(_type, _member) )  )
[   79s]       |                                ^~~~
[   79s] ib/ud/base/ud_ep.c:900:16: note: in expansion of macro 'ucs_queue_iter_elem'
[   79s]   900 |     sent_skb = ucs_queue_iter_elem(sent_skb, ep->resend.pos, queue);
[   79s]       |                ^~~~~~~~~~~~~~~~~~~
[   79s] ib/ud/base/ud_ep.c:912:56: error: taking address of packed member of 'struct <anonymous>' may result in an unaligned pointer value [-Werror=address-of-packed-member]
[   79s]   912 |         ep->resend.pos = ucs_queue_iter_next(ep->resend.pos);
[   79s]       |                                              ~~~~~~~~~~^~~~
[   79s] ib/ud/base/ud_ep.c:925:52: error: taking address of packed member of 'struct <anonymous>' may result in an unaligned pointer value [-Werror=address-of-packed-member]
[   79s]   925 |     ep->resend.pos = ucs_queue_iter_next(ep->resend.pos);
[   79s]       |                                          ~~~~~~~~~~^~~~
[   79s] libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -DCPU_FLAGS= -I/home/abuild/rpmbuild/BUILD/ucx-1.4.0/src -I/home/abuild/rpmbuild/BUILD/ucx-1.4.0/src -O3 -g -Wall -Werror -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -c sm/base/sm_iface.c  -fPIC -DPIC -o sm/base/.libs/libuct_la-sm_iface.o
[   79s] /bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..  -DCPU_FLAGS="" -I/home/abuild/rpmbuild/BUILD/ucx-1.4.0/src -I/home/abuild/rpmbuild/BUILD/ucx-1.4.0/src       -O3 -g -Wall -Werror  -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -c -o sm/base/libuct_la-sm_ep.lo `test -f 'sm/base/sm_ep.c' || echo './'`sm/base/sm_ep.c
[   79s] /bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..  -DCPU_FLAGS="" -I/home/abuild/rpmbuild/BUILD/ucx-1.4.0/src -I/home/abuild/rpmbuild/BUILD/ucx-1.4.0/src       -O3 -g -Wall -Werror  -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -c -o sm/mm/libuct_la-mm_iface.lo `test -f 'sm/mm/mm_iface.c' || echo './'`sm/mm/mm_iface.c
[   79s] libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -DCPU_FLAGS= -I/home/abuild/rpmbuild/BUILD/ucx-1.4.0/src -I/home/abuild/rpmbuild/BUILD/ucx-1.4.0/src -O3 -g -Wall -Werror -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -c sm/base/sm_ep.c  -fPIC -DPIC -o sm/base/.libs/libuct_la-sm_ep.o
[   79s] libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -DCPU_FLAGS= -I/home/abuild/rpmbuild/BUILD/ucx-1.4.0/src -I/home/abuild/rpmbuild/BUILD/ucx-1.4.0/src -O3 -g -Wall -Werror -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -c sm/mm/mm_iface.c  -fPIC -DPIC -o sm/mm/.libs/libuct_la-mm_iface.o
[   79s] sm/mm/mm_iface.c: In function 'uct_mm_iface_create_signal_fd':
[   79s] sm/mm/mm_iface.c:433:42: error: taking address of packed member of 'struct uct_mm_fifo_ctl' may result in an unaligned pointer value [-Werror=address-of-packed-member]
[   79s]   433 |                       (struct sockaddr *)&iface->recv_fifo_ctl->signal_sockaddr,
[   79s]       |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #3119

### CODE REVIEW

File: src/uct/sm/mm/base/mm_iface.h [hunk 1]
```diff
@@ -49,7 +49,7 @@ struct uct_mm_fifo_ctl {
 
     /* 2nd cacheline */
     volatile uint64_t  tail;       /* how much was read */
-} UCS_S_PACKED;
+} UCS_S_PACKED UCS_V_ALIGNED(UCS_SYS_CACHE_LINE_SIZE);
```
Comment (hoopoepg) [inline src/uct/sm/mm/base/mm_iface.h]: why not ```UCT_UD_SKB_ALIGN``` instead of ```UCS_V_ALIGNED(UCS_SYS_CACHE_LINE_SIZE)```?

Comment (yosefe) [reply src/uct/sm/mm/base/mm_iface.h]: this is MM transport.. also it relies on another macro which does cacheline padding..

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8541/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8547/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8551/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@hoopoepg can you pls review again? fixed ud

**Pull request merged**